integration/container/mounts_linux_test.go

```diff
@@ -82,3 +92,63 @@ func TestContainerShmNoLeak(t *testing.T) {
 		t.Fatalf("mount leaked: %s", string(out))
 	}
 }
+
+func TestContainerNetworkMountsNoChown(t *testing.T) {
+	// chown only applies to Linux bind mounted volumes; must be same host to verify
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux" || !testEnv.IsLocalDaemon())
+
+	defer setupTest(t)()
+
+	ctx := context.Background()
+
+	tmpDir, err := ioutils.TempDir("", "test-network-mounts")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	// make tmp dir readable by anyone to allow userns process to mount from
+	err = os.Chmod(tmpDir, 0755)
+	require.NoError(t, err)
+	// create temp files to use as network mounts
+	tmpNWFileMount := filepath.Join(tmpDir, "nwfile")
+
+	err = ioutil.WriteFile(tmpNWFileMount, []byte("network file bind mount"), 0644)
+	require.NoError(t, err)
+
+	config := container.Config{
+		Image: "busybox",
+	}
+	hostConfig := container.HostConfig{
+		Mounts: []mount.Mount{
+			{
+				Type:   "bind",
+				Source: tmpNWFileMount,
+				Target: "/etc/resolv.conf",
+			},
+			{
+				Type:   "bind",
+				Source: tmpNWFileMount,
+				Target: "/etc/hostname",
+			},
+			{
+				Type:   "bind",
+				Source: tmpNWFileMount,
+				Target: "/etc/hosts",
+			},
+		},
+	}
+
+	cli, err := client.NewEnvClient()
+	require.NoError(t, err)
+	defer cli.Close()
+
+	ctrCreate, err := cli.ContainerCreate(ctx, &config, &hostConfig, &network.NetworkingConfig{}, "")
+	require.NoError(t, err)
+	// container will exit immediately because of no tty, but we only need the start sequence to test the condition
+	err = cli.ContainerStart(ctx, ctrCreate.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	// check that host-located bind mount network file did not change ownership when the container was started
+	statT, err := system.Stat(tmpNWFileMount)
```

Comment:
Any idea why it would change ownership?

Reply:
@dnephin I think preserving the owner is the expected behavior. The test is to make sure no changes in owner.

Reply:
I understand that. I'm wondering how this test would ever fail. Was there some bug that caused it to be some other owner? What would it be other than root? Does this require userns or something to be of value?

Reply:
Looks like it was added in https://github.com/moby/moby/commit/42716dcf5c986e4cbb51f480f2782c05e5bd0b41, and to verify that in a non-usernamespace situation, these files don't change ownership.

Do you think we need to take the description from that commit and add it as description to this test case?

Reply:
@thaJeztah Added the PR #36226 in case we want to add the description.